File: EzRentAdmin/src/app/models/createCategoryWithParentReq.ts
import { Category } from './category';

export class CreateCategoryWithParentReq {
    username: string;
    password: string;
    newCategoryEntity: Category;
    parentCategoryId: number | undefined;

    constructor(username: string, password: string, newCategoryEntity: Category, parentCategoryId?: number) {
        this.username = username;
        this.password = password;
        this.newCategoryEntity = newCategoryEntity;
        this.parentCategoryId = parentCategoryId;
    }

    getUsername(): string {
        return this.username;
    }

    getPassword(): string {
        return this.password;
    }

    getNewCategoryEntity(): Category {
        return this.newCategoryEntity;
    }

    getParentCategoryId(): number | undefined {
        return this.parentCategoryId;
    }
}

File: EzRent/EzRentRws/src/java/ws/datamodel/CreateCategoryWithParentReq.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ws.datamodel;

import entity.CategoryEntity;

/**
 *
 * @author Yuxin
 */
public class CreateCategoryWithParentReq {
    private String username;
    private String password;
    private CategoryEntity newCategoryEntity;
    private Long parentCategoryId;
    
    public CreateCategoryWithParentReq() {
        
    }
    
    public CreateCategoryWithParentReq(String username, String password, CategoryEntity categoryEntity, Long parentCategoryId) {
        this.username = username;
        this.password = password;
        this.newCategoryEntity = categoryEntity;
        this.parentCategoryId = parentCategoryId;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public CategoryEntity getNewCategoryEntity() {
        return newCategoryEntity;
    }

    public void setNewCategoryEntity(CategoryEntity newCategoryEntity) {
        this.newCategoryEntity = newCategoryEntity;
    }

    public Long getParentCategoryId() {
        return parentCategoryId;
    }

    public void setParentCategoryId(Long parentCategoryId) {
        this.parentCategoryId = parentCategoryId;
    }
    
    
}
